### PR TITLE
Install the integration test trace listener in Roslyn OOP

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Workspaces/Remote/ServiceHub/WatsonTraceListener.cs [hunk 1]
```diff
@@ -6,11 +6,11 @@
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.ErrorReporting;
 
-namespace Microsoft.VisualStudio.IntegrationTest.Setup
+namespace Microsoft.CodeAnalysis.Remote
 {
```
Comment (tmat) [inline src/Workspaces/Remote/ServiceHub/WatsonTraceListener.cs:10]: I'd put this to Microsoft.CodeAnalysis.ErrorReporting namespace

File: src/VisualStudio/Core/Def/Implementation/Watson/WatsonTraceListener.cs [hunk 2]
```diff
@@ -4,13 +4,10 @@
...
 
-namespace Microsoft.VisualStudio.IntegrationTest.Setup
+namespace Microsoft.CodeAnalysis.ErrorReporting
 {
-    using Debugger = System.Diagnostics.Debugger;
-
-    internal class IntegrationTestTraceListener : TraceListener
+    internal class WatsonTraceListener : TraceListener
     {
         public override void Fail(string message, string detailMessage)
```
Comment (tmat) [inline src/VisualStudio/Core/Def/Implementation/Watson/WatsonTraceListener.cs:12]: >Fail [](start = 29, length = 4)

How does this behave when the debugger is attached? When debugging we shouldn't just fail fast on failed assert.

Comment (sharwell) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonTraceListener.cs:12]: This code just delegates to `FatalError.Report`. Handling for debugger attachment, if any, would be there.

Also note that attaching the debugger to the ServiceHub process is not a standard scenario.

Comment (tmat) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonTraceListener.cs:12]: > Also note that attaching the debugger to the ServiceHub process is not a standard scenario.

I do it all the time. How else would you debug our remote services?

**Pull request merged**